freedomofpress/securedrop-workstation
# Remove legacy updater

## Status

Ready for review

## Description

Resolves #412

The preflight updater is now at feature parity with the cron job that fires every 24 hours to update VMs:
- We enforce updates before launching the client (#396)
- We remind the user if updates have not been applied recently (#445)
- We apply the dom0 state as part of the update process (#458)

Therefore, it is safe to remove the cron job at this time. This PR preserves the `securedrop-login` script, which enforces updates to the `sd-viewer` template on login, until #415 is resolved.

**NOTE:** Because it changes the contents of the RPM package, this PR bumps the RPM version.

## Test plan
### Preparatory steps

- Make sure you are using a workstation that has been previously provisioned from a different branch that includes the updater (e.g., `master`)
- Check out this branch in your dev VM, `make clone` it into `dom0`

### Test plan 
- Run `make clean`
- Confirm the following files do _not_ exist (`make clean` logic is still removing them)
  - [ ] `/srv/salt/securedrop-update`
  - [ ] `/usr/bin/securedrop-update`
  - [ ] `/etc/cron.daily/securedrop-update-cron` (symlink)
- Run `make all`
- Confirm the following files do _not_ exist (`make clean` logic is still removing them)
  - [ ] `/srv/salt/securedrop-update`
  - [ ] `/usr/bin/securedrop-update`
  - [ ] `/etc/cron.daily/securedrop-update-cron` (symlink)

## Checklist
- [x] `make flake8` passes
- [ ] `make test` passes in dom0
    - I am seeing mismatches in the RPC policy test, which appear to be unrelated to this PR, which does not touch RPC policies.
- [x] I have added/removed files, and have updated packaging logic in MANIFEST.in and rpm-build/SPECS/securedrop-workstation-dom0-config.spec

## Attribution

This PR is based on #322, originally by @kushaldas.
